show_graph.py

```diff
@@ -21,44 +18,62 @@ def DrawBarGraph(connection, table_name, y_axis_field, x_axis_values,
     x_axis_values: the values in x axis to compare
     x_axis_field: column name in the table for the x axis
   """
-  direct_lists = list()
-  envoy_lists = list()
-
-  for x in x_axis_values:
-    condition = ("where {}=\"{}\" and"
-                 " category=\"direct\"").format(x_axis_field, x)
-    direct_list = db_utils.SingleColumnToList(db_utils.GetFieldFromTable(
-        connection, table_name, field=y_axis_field, cond=condition))
-    if not direct_list:
-      print "{} {} is not found in table for direct results.".format(
-          x_axis_field, x)
-      direct_list = [0]
-
-    direct_lists.append(direct_list)
-
-    condition = ("where {}=\"{}\" and"
-                 " category=\"envoy\"").format(x_axis_field, x)
-    envoy_list = db_utils.SingleColumnToList(db_utils.GetFieldFromTable(
-        connection, table_name, field=y_axis_field, cond=condition))
-    if not envoy_list:
-      print "{} {} is not found in table for Envoy results.".format(
-          x_axis_field, x)
-      envoy_list = [0]
-
-    envoy_lists.append(envoy_list)
+
+  def GetListsFromDB(x_axis_values, x_axis_field, connection,
+                     table_name, y_axis_field, category):
+    """This function returns lists of values of a field from the DB.
```

Comment:
No need to change anything, but for future reference, there's no need to write details function-level comments for internal functions unless it's really unclear what's going on - you can just add a few line # comment above explaining what it does. The full arg/returns docs are for module-level functions that are likely to be exported out and consumed by others.